presto-main/src/main/java/com/facebook/presto/execution/scheduler/SqlQueryScheduler.java

```diff
@@ -582,16 +589,38 @@ else if (partitioningHandle.equals(SCALED_WRITER_DISTRIBUTION)) {
                     getWriterMinSize(session));
             whenAllStages(childStageExecutions, StageExecutionState::isDone)
                     .addListener(scheduler::finish, directExecutor());
-            stageSchedulers.put(stageId, scheduler);
         }
+        return stageScheduler;
```

Comment:
The scheduler for `SCALED_WRITER_DISTRIBUTION` is never assigned and returned.

Reply:
Let's do `if (partitioningHandle.equals(SCALED_WRITER_DISTRIBUTION)) {` as another `else if`